src/classes/RD2_NamingService.cls

```diff
@@ -0,0 +1,206 @@
+        Boolean hasChanged = rd.npe03__Organization__c != oldRd.npe03__Organization__c
+            || rd.npe03__Contact__c != oldRd.npe03__Contact__c
+            || rd.npe03__Amount__c != oldRd.npe03__Amount__c
+            || rd.Name == System.Label.npo02.NameReplacementText;
+
+        if (!hasChanged && isMultiCurrencyEnabled) {
+            hasChanged = (String) rd.get(CURRENCY_ISO_CODE_FIELD) != (String) oldRd.get(CURRENCY_ISO_CODE_FIELD);
+        }
+
+        return hasChanged;
+    }
+
+    /***
+    * @description Autogenerates recurring donation names for all specified records
+    * @param rds New Recurring Donation records
+    * @return none
+    */
+    public void autogenerateNames(List<npe03__Recurring_Donation__c> rds) {
+        if (!isAutomaticNamingEnabled || rds == null || rds.isEmpty()) {
+            return;
+        }
+
+        Map<Id, RD2_Donor.Record> donorById = donorGateway.getDonors(rds);
+
+        for (npe03__Recurring_Donation__c rd : rds) {
+            rd.Name = constructName(rd, donorById);
+        }
+    }   
+
+    /***
+    * @description Constructs recurring donation name
+    * Name formatting: 
+    *   If multi-currency, then "{!DonorName} {ISO Code} {!Amount} - Recurring".
+    *   Otherwise, "{!DonorName} {User Default Currency Symbol}{!Amount} - Recurring".
+    *
+    * @param rd Recurring Donation record
+    * @param donorById Account/Contact donor record by Id
+    * @return String New name for the Recurring Donation record
+    */
+    private String constructName(npe03__Recurring_Donation__c rd,  Map<Id, RD2_Donor.Record> donorById) {
+        RD2_Donor.Record donor = rd.npe03__Contact__c != null
+            ? donorById.get(rd.npe03__Contact__c)
+            : donorById.get(rd.npe03__Organization__c);
+
+        String nameSuffix = ' ' + formatAmount(rd) + ' - ' + System.Label.RecurringDonationNameSuffix;
+
+        String donorName = donor == null 
+            ? '' 
+            : donor.Name.abbreviate(MAX_NAME_LENGTH - nameSuffix.length());
+
+        return donorName + nameSuffix;
+    }
+
+    /***
+    * @description Formats amount used to construct the name    
+    * @param rd Recurring Donation record
+    * @return String Formatted amount
+    */
+    private String formatAmount(npe03__Recurring_Donation__c rd) {
+        String formattedAmount = rd.npe03__Amount__c == null ? '' : rd.npe03__Amount__c.format();
+
+        if (formattedAmount.endsWith('.00')) {
+            formattedAmount = formattedAmount.removeEnd('.00');
+        } else if (formattedAmount.endsWith('.0')) {
+            formattedAmount = formattedAmount.removeEnd('.0');
+        }
+
+        if (isMultiCurrencyEnabled) {
+            String code = (String) rd.get(CURRENCY_ISO_CODE_FIELD);
+
+            if (String.isNotBlank(code)) {
+                formattedAmount = code + ' ' + formattedAmount;
+            }
+        } else {
+            formattedAmount = userDefaultCurrencySymbol + formattedAmount;
```

Comment:
@RadSFDeveloper: @Julian88Tex made a very good suggestion. We made this change to the OppNaming service to add a space between the _symbol_ and the amount when the symbol is more than 1 character in length. It very much helps with readability. @Julian88Tex recommended that we make the same change here as well. Here's an example of what it could generate with and without the space. The second bullet is much easier to read than the first.
- Without: ₨100
- With: ₨ 100

```suggestion
            if (!String.isBlank(userDefaultCurrencySymbol) && userDefaultCurrencySymbol.length() > 1) {
                formattedAmount = userDefaultCurrencySymbol + ' ' + formattedAmount;
            } else {
                formattedAmount = userDefaultCurrencySymbol + formattedAmount;
            }
```

Reply:
@force2b , the change has been applied. Can you please review again (note package.xml is fixed by `cci` deploy).
@sam-knox , can you please review PR's Changes above and check if they need to be modified based on the latest requirement?